include/arch/arm/aarch32/cortex_m/scripts/linker.ld

```diff
@@ -98,11 +98,33 @@ MEMORY
     DT_REGION_FROM_NODE_STATUS_OKAY(SRAM1, rw, DT_NODELABEL(sram1))
     DT_REGION_FROM_NODE_STATUS_OKAY(SRAM2, rw, DT_NODELABEL(sram2))
     /* STM32 alternate RAM configurations */
+
+#if DT_NODE_HAS_STATUS(DT_ALIAS(sramnocache), okay)
+    DT_REGION_FROM_NODE_STATUS_OKAY(sramnocache, rw, DT_ALIAS(sramnocache))
```

Comment:
It appears that the purpose of these changes is purely to rename the memory region of whichever node `sramnocache` points to to `sramnocache`. This adds a lot of conditional and duplication to achieve this, and will become unreadable as soon as another node also needs to have its name changed.

Reply:
Well, some knowledge in preprocessor and linker skripts is already assumend. But my implementation is simpliest possible for this purpose.

Reply:
Unreadable from the perspective of the number of conditionals that need to be added.
This only looks "simple" because this is the first special case. When someone wants to add `DT_ALIAS(sramcache)` that changes the name of a different SRAM region, you end up in a case of `2^n` combinations.
I disagree that this is the simplest way possible to change the name of a single memory region.

Reply:
Do You have then any better ideas?

Reply:
Yes, the PR which mentions this one, #36196.
This allows both changing the memory region name, and automatically extracting the region name from a `chosen` or `alias` node.

Reply:
Ok. Thanks for that. I see your point. Of course, using some script is more elegant solution, than using ifdefs in linker skript. Moreover it definitely increase readability.
Anyway I still would prefer defining special use cases like none-cacheable memory regions by fixed aliases instead of fixed label names. The reason for that is that, labels may be changed b the user and used for other stuff than linker script. Thus your solution fixes label names, which is not completely cosy to use. I think, if you would add aliases in your PR it would be best one solution based on good readable script and you would not need to change lable names in multiple boards. 
For example: Adding aliases like sramnocache256b, sramnocache16k - would increase usability.

Reply:
I have no problem with using fixed aliases instead of fixed label names, in fact I think that is definitely what you should do.
That the memory region name is generated from the label is beside the point. The point is that there is a defined mapping from a devicetree node to the name of the region generated by that node.

As I noted above,  `GROUP_DATA_LINK_IN(sramnocache, sramnocache)` becomes `GROUP_DATA_LINK_IN(DT_LABEL(DT_ALIAS(sramnocache)),DT_LABEL(DT_ALIAS(sramnocache)))`. At that point you can change the label name to `ALEXANDERS_SRAM` and everything will still work fine.